### PR TITLE
Code style the files

### PR DESCRIPTION
**Pull request by danewalton-msft**
- Code style the files
- Adds a gate check for code style

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,6 +17,24 @@ jobs:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
 
+      - name: Style Check
```
Comment (vaavva) [inline .github/workflows/main.yml:27]: Can you put info on how to run this manually in the contributing guidelines as well?

File: examples/Azure_IoT_Adu_ESP32/Azure_IoT_Adu_ESP32.ino [hunk 2]
```diff
@@ -138,87 +140,78 @@ static AzIoTSasToken sasToken(
...
+        0x0b, 0x35, 0xdc, 0x73, 0xc8, 0x69, 0xd0, 0x5e, 0xe8, 0x25, 0x43, 0x9e, 0xf6, 0xd8, 0xab,
+        0x05, 0xaf, 0x51, 0x29, 0x23, 0x55, 0x40, 0x58, 0x10, 0xea, 0xb8, 0xe2, 0xcd, 0x5d, 0x79,
+        0xcc, 0xec, 0xdf, 0xb4, 0x5b, 0x98, 0xc7, 0xfa, 0xe3, 0xd2, 0x6c, 0x26, 0xce, 0x2e, 0x2c,
+        0x56, 0xe0, 0xcf, 0x8d, 0xee, 0xfd, 0x93, 0x12, 0x2f, 0x00, 0x49, 0x8d, 0x1c, 0x82, 0x38,
+        0x56, 0xa6, 0x5d, 0x79, 0x44, 0x4a, 0x1a, 0xf3, 0xdc, 0x16, 0x10, 0xb3, 0xc1, 0x2d, 0x27,
+        0x11, 0xfe, 0x1b, 0x98, 0x05, 0xe4, 0xa3, 0x60, 0x31, 0x99 };
+static uint8_t ucAzureIoTADURootKeyE200703[3] = { 0x01, 0x00, 0x01 };
+
+static SampleJWS::RootKey xADURootKeys[]
+    = { { // Minus one on id to not count NULL
```
Comment (vaavva) [inline examples/Azure_IoT_Adu_ESP32/Azure_IoT_Adu_ESP32.ino:205]: nit: This is kinda funky here - shouldn't the new line for `{` catch this?

File: examples/Azure_IoT_Central_ESP32/Azure_IoT_PnP_Template.cpp [hunk 3]
```diff
@@ -91,23 +91,25 @@ static bool led2_on = false;
...
-void azure_pnp_init()
-{
-}
+void azure_pnp_init() {}
 
-const az_span azure_pnp_get_model_id()
-{
-  return AZ_SPAN_FROM_STR(AZURE_PNP_MODEL_ID);
-}
+const az_span azure_pnp_get_model_id() { return AZ_SPAN_FROM_STR(AZURE_PNP_MODEL_ID); }
```
Comment (vaavva) [inline examples/Azure_IoT_Central_ESP32/Azure_IoT_PnP_Template.cpp:112]: nit: Here and right above also seem to have some funky braces compared to what we usually do. (This set feels less weird than the one above though if it's not an easy fix or it is the expected behavior)

**Review by vaavva — APPROVED** [review]
Looked through the ADU ESP32, Central ESP32, ESP8266, and RealtekAmeba samples to try and look over at least one of each type of file for the auto changes - only weird thing is sometimes it's using `{` without newlines (commented on a few) and I wasn't sure if that was expected or not, but everything else looks good!

**Comment by danewalton-msft** [comment]
> Looked through the ADU ESP32, Central ESP32, ESP8266, and RealtekAmeba samples to try and look over at least one of each type of file for the auto changes - only weird thing is sometimes it's using `{` without newlines (commented on a few) and I wasn't sure if that was expected or not, but everything else looks good!

Yea I think double braces like the one you pointed out might get formatted slightly weird. Single line functions also seem to be put on a single line instead of always doing a newline break for the curly braces (as long as they're under a certain character count).

Some might be weird but I'm inclined to leave it to run its course and be the single point of truth. Since this is copied exactly as the embedded C does it.

**Pull request merged**